# Mechanisms of Action (MoA) Prediction

Can you improve the algorithm that classifies drugs based on their biological activity?

In [29]:
import pandas as pd

In [6]:
%%sh 
ls data/

sample_submission.csv
test_features.csv
train_features.csv
train_targets_nonscored.csv
train_targets_scored.csv


In [12]:
df_test_features = pd.read_csv("data/test_features.csv")
df_train_features = pd.read_csv("data/train_features.csv")
df_train_targets_nonscored = pd.read_csv("data/train_targets_nonscored.csv")
df_train_targets_scored = pd.read_csv("data/train_targets_scored.csv")

## train features

Features for the training set. Features g- signify gene expression data, and c- signify cell viability data. cp_type indicates samples treated with a compound (cp_vehicle) or with a control perturbation (ctrl_vehicle); control perturbations have no MoAs; cp_time and cp_dose indicate treatment duration (24, 48, 72 hours) and dose (high or low), sig_id = sample id

In [13]:
df_train_features.head()

sig_id cp_type  cp_time cp_dose     g-0     g-1     g-2     g-3  \
0  id_000644bb2  trt_cp       24      D1  1.0620  0.5577 -0.2479 -0.6208   
1  id_000779bfc  trt_cp       72      D1  0.0743  0.4087  0.2991  0.0604   
2  id_000a6266a  trt_cp       48      D1  0.6280  0.5817  1.5540 -0.0764   
3  id_0015fd391  trt_cp       48      D1 -0.5138 -0.2491 -0.2656  0.5288   
4  id_001626bd3  trt_cp       72      D2 -0.3254 -0.4009  0.9700  0.6919   

      g-4     g-5  ...    c-90    c-91    c-92    c-93    c-94    c-95  \
0 -0.1944 -1.0120  ...  0.2862  0.2584  0.8076  0.5523 -0.1912  0.6584   
1  1.0190  0.5207  ... -0.4265  0.7543  0.4708  0.0230  0.2957  0.4899   
2 -0.0323  1.2390  ... -0.7250 -0.6297  0.6103  0.0223 -1.3240 -0.3174   
3  4.0620 -0.8095  ... -2.0990 -0.6441 -5.6300 -1.3780 -0.8632 -1.2880   
4  1.4180 -0.8244  ...  0.0042  0.0048  0.6670  1.0690  0.5523 -0.3031   

     c-96    c-97    c-98    c-99  
0 -0.3981  0.2139  0.3801  0.4176  
1  0.1522  0.1241  0.6077  0.7371  
2 -0.6417 -0.2187 -1.4080  0.6931  
3 -1.6210 -0.8784 -0.3876 -0.8154  
4  0.1094  0.2885 -0.3786  0.7125  

[5 rows x 876 columns]

In [14]:
df_train_features.info()

&lt;class &#39;pandas.core.frame.DataFrame&#39;&gt;
RangeIndex: 23814 entries, 0 to 23813
Columns: 876 entries, sig_id to c-99
dtypes: float64(872), int64(1), object(3)
memory usage: 159.2+ MB


In [22]:
df_train_features.shape

(23814, 876)

In [51]:
len([x for x in list(df_train_features.columns) if x[0:2] == 'g-'])

772

In [53]:
len([x for x in list(df_train_features.columns) if x[0:2] == 'c-'])

100

772 gene expression columns, 100 c- signify cell viability columns, 'sig_id', 'cp_type', 'cp_time', 'cp_dose'